packages/eslint-plugin/src/utils/runtime-lint-utils.ts

```diff
@@ -463,28 +463,31 @@ export function groupImports(
 }
 
 /**
- * Checks if import points to a secondary entry point in Angular project
- * @param targetProjectLocator
- * @param importExpr
- * @returns
+ * Checks if source file belongs to a secondary entry point different than the import one
  */
-export function isAngularSecondaryEntrypoint(
+export function belongsToDifferentNgEntryPointpnpm(
   importExpr: string,
   filePath: string,
   projectRoot: string
 ): boolean {
-  const resolvedModule = resolveModuleByImport(
+  const resolvedImportFile = resolveModuleByImport(
     importExpr,
-    filePath,
+    filePath, // not strictly necessary, but speeds up resolution
     join(workspaceRoot, getRootTsConfigFileName())
   );
 
-  return (
-    !!resolvedModule && fileIsSecondaryEntryPoint(resolvedModule, projectRoot)
-  );
+  if (!resolvedImportFile) {
+    return false;
+  }
+
+  const importEntryPoint = getEntryPoint(resolvedImportFile, projectRoot);
+  const srcEntryPoint = getEntryPoint(filePath, projectRoot);
+
+  // check if the entry point of import expression is different than the source file's entry point
+  return importEntryPoint !== srcEntryPoint;
 }
 
-function fileIsSecondaryEntryPoint(file: string, projectRoot: string): boolean {
+function getEntryPoint(file: string, projectRoot: string): string {
```

Comment:
Shall we also add Ng here to try and make it explicit its Angular specific?

Reply:
Yeah, that's a good suggestion.